featureflags.go

```diff
@@ -638,7 +640,7 @@ func (poller *FeatureFlagsPoller) matchFeatureFlagProperties(
 	}
 
 	if isInconclusive {
-		return false, &InconclusiveMatchError{"Can't determine if feature flag is enabled or not with given properties"}
+		return nil, &InconclusiveMatchError{"Can't determine if feature flag is enabled or not with given properties"}
```

Comment:
why did this return type change?

Reply:
There's other code paths in this function which also return `nil` on error.
https://github.com/PostHog/posthog-go/blob/e529d35d7cef1c374bd2519b8a4ec75b62730bdd/featureflags.go#L614-L627

What'd I'd observed was that in certain cases it'd be possible to get `false, Error{}` back from `getFeatureFlags`, so this change was to normalize that response to always be `nil, Error{}` if it was inconclusive.

Though, I do want to circle back on this before merging. I did rebase before pushing and it's possible that this may have changed.